model/lifecycle.go

```diff
@@ -1297,6 +1267,18 @@ func createOneTask(id string, buildVarTask BuildVariantTaskUnit, project *Projec
 		// TODO (EVG-16169): set ExecutionPlatform based on whether or not the task
 		// is running in a host or container.
 		ExecutionPlatform: task.ExecutionPlatformHost,
+		RunOnContainer:    utility.FromBoolPtr(buildVarTask.RunOnContainer),
+	}
+	flags, err := evergreen.GetServiceFlags()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting service flags")
+	}
+	if !flags.ContainerConfigurationsDisabled && utility.FromBoolPtr(buildVarTask.RunOnContainer) {
```

Comment:
I think if container configurations are disabled, this should error and fail to allow the task to be created. Otherwise, if they're trying to use containers and it's disabled by this flag, they'll end up with an invalid task associated with a nonexistent distro which can't run anyways.